st2common/st2common/config.py

```diff
@@ -374,6 +374,10 @@ def register_opts(ignore_errors=False):
             'virtualenv_opts', default=['--system-site-packages'],
             help='List of virtualenv options to be passsed to "virtualenv" command that '
                  'creates pack virtualenv.'),
+        cfg.ListOpt(
+            'pip_opts', default=[],
+            help='List of pip options to be passed to "pip install" command when installing pack '
+                 'dependencies into pack virtual environment.'),
```

Comment:
What do these pip related changes have to do with the changes to not directly call eventlet?

Reply:
Ideally I would open separate PR for each of those set of changes, but I kinda wanted to avoid that.